src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsImmediateStatementCompletion2.cs

```diff
@@ -68,7 +71,36 @@ int IVsImmediateStatementCompletion2.InstallStatementCompletion(int install, IVs
 
         int IVsImmediateStatementCompletion2.SetCompletionContext(string filePath,
             IVsTextLines buffer,
-            Microsoft.VisualStudio.TextManager.Interop.TextSpan[] currentStatementSpan,
+            TextSpan[] currentStatementSpan,
+            object punkContext,
+            IVsTextView textView)
+        {
+            var waitIndicator = this.Package.ComponentModel.GetService<IWaitIndicator>();
+            waitIndicator.Wait(ServicesVSResources.Debugger, ServicesVSResources.Loading_immediate_window,
+                allowCancel: false,
+                action: waitContext =>
+                {
+                    var bufferAndContext = GetCompletionContextAsync(filePath, buffer, currentStatementSpan, punkContext, textView).WaitAndGetResult(waitContext.CancellationToken);
+                    if (bufferAndContext.Item1 != null)
+                    {
+                        if (bufferAndContext.Item2 != null)
+                        {
+                            this.filters[textView].SetContext(bufferAndContext.Item2);
+                        }
+                        else
+                        {
+                            this.filters[textView].RemoveContext();
+                        }
+                    }
+                });
+
+            return VSConstants.S_OK;
+        }
+
+        private async Task<ValueTuple<IVsTextLines, AbstractDebuggerIntelliSenseContext>> GetCompletionContextAsync(
```

Comment:
Does this need to return a tuple? The caller doesn't use the returned IVsTextLines.